stock_picking_comment_template/tests/test_stock_picking_report.py

```diff
@@ -5,55 +5,53 @@
 
 
 class TestStockPickingReport(TransactionCase):
-    def setUp(self):
-        super().setUp()
-        self.company = self.env.ref("base.main_company")
-        self.base_comment_model = self.env["base.comment.template"]
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.company = cls.env.ref("base.main_company")
+        cls.base_comment_model = cls.env["base.comment.template"]
         # Create comment related to sale model
-        self.picking_obj = self.env.ref("stock.model_stock_picking")
-        self.before_comment = self._create_comment(self.picking_obj, "before_lines")
-        self.after_comment = self._create_comment(self.picking_obj, "after_lines")
+        cls.picking_obj = cls.env.ref("stock.model_stock_picking")
+        cls.before_comment = cls._create_comment_template_test(cls, "before_lines")
+        cls.after_comment = cls._create_comment_template_test(cls, "after_lines")
         # Create partner
-        self.partner = self.env["res.partner"].create({"name": "Partner Test"})
-        self.partner.base_comment_template_ids = [
-            (4, self.before_comment.id),
-            (4, self.after_comment.id),
+        cls.partner = cls.env["res.partner"].create({"name": "Partner Test"})
+        cls.partner.base_comment_template_ids = [
+            (4, cls.before_comment.id),
+            (4, cls.after_comment.id),
         ]
-        self.picking_model = self.env["stock.picking"]
-        self.picking = self.picking_model.create(
+        cls.picking_model = cls.env["stock.picking"]
+        cls.picking = cls.picking_model.create(
             {
-                "partner_id": self.partner.id,
-                "location_id": self.ref("stock.stock_location_stock"),
-                "location_dest_id": self.ref("stock.stock_location_customers"),
-                "picking_type_id": self.ref("stock.picking_type_out"),
+                "partner_id": cls.partner.id,
+                "location_id": cls.env.ref("stock.stock_location_stock").id,
+                "location_dest_id": cls.env.ref("stock.stock_location_customers").id,
+                "picking_type_id": cls.env.ref("stock.picking_type_out").id,
             }
         )
 
-    def _create_comment(self, model, position):
+    def _create_comment_template_test(self, position):
```

Comment:
Is it necessary to change the name and arguments of this method?

Reply:
Renamed to the original. Fields have been simplified since by passing self you can now acquire the necessary field from there.